src/java.base/share/classes/com/sun/crypto/provider/ChaCha20Cipher.java

```diff
@@ -226,6 +226,11 @@ protected int engineGetOutputSize(int inputLen) {
     protected AlgorithmParameters engineGetParameters() {
         AlgorithmParameters params = null;
         if (mode == MODE_AEAD) {
+            // In a pre-initialized state or any state without a nonce value
+            // this call should cause a random nonce to be generated.
+            if (!initialized || nonce == null) {
+                nonce = createRandomNonce(null);
```

Comment:
Should nonce be a local variable instead? I think you don't want the nonce field to be set unless the caller passes back in the returned params in an init call.

Reply:
Yes, it should be a local variable.  I'll fix this.